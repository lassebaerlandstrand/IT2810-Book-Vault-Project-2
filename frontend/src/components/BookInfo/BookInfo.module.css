.titleContainer {
  width: 95%;
}

@media screen and (max-width: $mantine-breakpoint-sm) {
  .titleContainer {
    width: auto;
  }
}
