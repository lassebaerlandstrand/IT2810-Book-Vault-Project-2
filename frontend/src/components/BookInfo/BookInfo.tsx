import { Container, Flex, Grid, Group, Image, Rating, Spoiler, Text } from '@mantine/core';
import { Book } from '@/generated/graphql';
import InfoGrid from '../InfoGrid/InfoGrid';
import styles from './BookInfo.module.css';

type BookInfoProps = {
  book: Book;
};

const BookInfo = ({ book }: BookInfoProps) => {
  return (
    <Group justify="center" gap="lg">
      <Container p="xs" className={styles.titleContainer}>
        <Text size="lg" fw={700} component="h1">
          {book.title}
        </Text>
        <Text size="md" c="dimmed">
          {book.authors[0]} {book.authors.length > 1 ? 'et al.' : ''}
        </Text>
      </Container>
      <Grid justify="left" align="top">
        <Grid.Col span={{ xs: 12, sm: 4, md: 4, lg: 4, xl: 4 }} h={400}>
          <Image
            src={book.coverImg}
            alt={`Cover image for ${book.title}`}
            fit="contain"
            fallbackSrc="https://placehold.co/200x300?text=Cover%20image%20for%20book"
            radius="lg"
            sizes="xs"
            w="fit-content"
            maw="100%"
            mah="100%"
            m="auto"
          />
        </Grid.Col>
        <Grid.Col span="auto">
          <Flex gap="sm" justify="center" align="center" direction="row" wrap="wrap">
            <Container p="xs" size="sm">
              <Text>
                {book.genres.slice(0, 3).join(', ')}
                {book.genres.length > 3 ? '...' : null}
              </Text>

              <Flex justify="center" align="center" gap={7} mt="xs">
                <Rating value={Math.round(book.rating * 2) / 2} fractions={2} readOnly />
                <Text fw={500}>{book.rating.toFixed(1)}</Text>
              </Flex>
            </Container>

            <Spoiler maxHeight={250} hideLabel={'Show less'} showLabel={'Show more'}>
              <Text>{book.description}</Text>
            </Spoiler>
          </Flex>
        </Grid.Col>

        <InfoGrid book={book} />
      </Grid>
    </Group>
  );
};

export default BookInfo;
