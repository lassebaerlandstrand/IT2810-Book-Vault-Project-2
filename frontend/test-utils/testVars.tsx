import { Book } from '@/generated/graphql';

export const dummyBook: Book = {
  __typename: 'Book',
  authors: [{ __typename: 'Author', name: 'A name' }],
  coverImg: 'https://placehold.co/200x300?text=Cover%20image%20for%20book',
  genres: [{ __typename: 'Genre', name: 'A genre' }],
  id: '1',
  publisher: { __typename: 'Publisher', name: 'A publisher' },
  title: 'A title',
  bookFormat: 'Paper',
  characters: ['A character'],
  description: 'A deep book about deep topics',
  isbn: '999999999',
  language: 'Swedish',
  pages: 1000,
  publishDate: '03/03/03',
  rating: 3.555,
  series: 'Title #1',
  numRatings: 5000,
  ratingsByStars: [1, 5, 10, 7, 1],
};
