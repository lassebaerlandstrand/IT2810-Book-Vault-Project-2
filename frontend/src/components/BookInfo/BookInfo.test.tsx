import { render, screen } from '@test-utils';
import { dummyBook } from '../../../test-utils/testVars';
import BookInfo from './BookInfo';

describe('BookInfo component', () => {
  test('renders the rating properly', () => {
    render(<BookInfo book={dummyBook} />);

    expect(screen.getAllByText(dummyBook.rating.toFixed(1))).toHaveLength(2);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<BookInfo book={dummyBook} />);
    const attributesToRemove = [
      ...document.body.querySelectorAll('div [id^="mantine"]'),
      ...document.body.querySelectorAll('div [for^="mantine"]'),
    ]; // Because Mantine uses random ids which causes snapshots to fail
    attributesToRemove.forEach((element) => {
      element.removeAttribute('for');
      element.removeAttribute('id');
      element.removeAttribute('aria-describedby');
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
