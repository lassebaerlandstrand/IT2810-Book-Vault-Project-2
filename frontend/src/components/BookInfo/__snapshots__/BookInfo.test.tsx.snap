// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BookInfo component > matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: Poppins, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji;--mantine-font-family-headings: Poppins, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji;--mantine-breakpoint-xs: 30em;--mantine-breakpoint-md: 64em;--mantine-breakpoint-xl: 90em;--mantine-breakpoint-xxs: 24em;--mantine-color-logo-theme-0: #fefefe;--mantine-color-logo-theme-1: #cbcfd2;--mantine-color-logo-theme-2: #97a0a6;--mantine-color-logo-theme-3: #7d8991;--mantine-color-logo-theme-4: #63717b;--mantine-color-logo-theme-5: #495a65;--mantine-color-logo-theme-6: #2f434f;--mantine-color-logo-theme-7: #132937;--mantine-color-logo-theme-8: #0b202e;--mantine-color-logo-theme-9: #061017;}:root[data-mantine-color-scheme="dark"]{--mantine-color-logo-theme-text: var(--mantine-color-logo-theme-4);--mantine-color-logo-theme-filled: var(--mantine-color-logo-theme-8);--mantine-color-logo-theme-filled-hover: var(--mantine-color-logo-theme-9);--mantine-color-logo-theme-light: rgba(47, 67, 79, 0.15);--mantine-color-logo-theme-light-hover: rgba(47, 67, 79, 0.2);--mantine-color-logo-theme-light-color: var(--mantine-color-logo-theme-3);--mantine-color-logo-theme-outline: var(--mantine-color-logo-theme-4);--mantine-color-logo-theme-outline-hover: rgba(99, 113, 123, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-logo-theme-text: var(--mantine-color-logo-theme-filled);--mantine-color-logo-theme-filled: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-filled-hover: var(--mantine-color-logo-theme-7);--mantine-color-logo-theme-light: rgba(47, 67, 79, 0.1);--mantine-color-logo-theme-light-hover: rgba(47, 67, 79, 0.12);--mantine-color-logo-theme-light-color: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-outline: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-outline-hover: rgba(47, 67, 79, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 29.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 30em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 63.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 64em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 89.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 90em) {.mantine-hidden-from-xl {display: none !important;}}@media (max-width: 23.99375em) {.mantine-visible-from-xxs {display: none !important;}}@media (min-width: 24em) {.mantine-hidden-from-xxs {display: none !important;}}
  </style>
  <div
    class="m_4081bf90 mantine-Group-root"
    style="--group-gap: var(--mantine-spacing-lg); --group-align: center; --group-justify: center; --group-wrap: wrap;"
  >
    <div
      class="m_7485cace mantine-Container-root"
    >
      <h1
        class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
        data-size="lg"
        style="--text-fz: var(--mantine-font-size-lg); --text-lh: var(--mantine-line-height-lg); font-weight: 700;"
      >
        A title
      </h1>
      <p
        class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
        data-size="md"
        style="--text-fz: var(--mantine-font-size-md); --text-lh: var(--mantine-line-height-md);"
      >
        A name 
      </p>
    </div>
    <div
      class="m_8bffd616 mantine-Flex-root __m__-r2r"
      style="gap: var(--mantine-spacing-sm); align-items: center; justify-content: center; flex-wrap: wrap; flex-direction: row;"
    >
      <img
        alt="Cover image for A title"
        class="m_9e117634 mantine-Image-root"
        sizes="xs"
        src="https://placehold.co/200x300?text=Cover%20image%20for%20book"
        style="--image-radius: var(--mantine-radius-lg); --image-object-fit: contain;"
      />
      <div
        class="m_8bffd616 mantine-Flex-root __m__-r2u"
        style="gap: var(--mantine-spacing-sm); align-items: center; justify-content: center; flex-wrap: wrap; flex-direction: row;"
      >
        <div
          class="m_7485cace mantine-Container-root"
          data-size="sm"
          style="--container-size: 48em;"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            [object Object]
          </p>
          <div
            class="m_8bffd616 mantine-Flex-root __m__-r32"
            style="gap: calc(0.4375rem * var(--mantine-scale)); align-items: center; justify-content: center;"
          >
            <div
              class="m_f8d312f2 mantine-Rating-root"
              data-size="sm"
              style="--rating-size: var(--rating-size-sm); --rating-color: var(--mantine-color-yellow-filled);"
            >
              <div
                class="m_61734bb7 mantine-Rating-symbolGroup"
              >
                <div
                  class="m_21342ee4 mantine-Rating-label"
                  data-read-only="true"
                  style="--rating-item-z-index: 0;"
                >
                  <div
                    class="m_fae05d6a mantine-Rating-symbolBody"
                    style="--rating-symbol-clip-path: inset(0 100% 0 0);"
                  >
                    <svg
                      class="m_5662a89a mantine-Rating-starSymbol"
                      data-filled="true"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="m_21342ee4 mantine-Rating-label"
                  data-read-only="true"
                  style="--rating-item-z-index: 0;"
                >
                  <div
                    class="m_fae05d6a mantine-Rating-symbolBody"
                    style="--rating-symbol-clip-path: inset(0 100% 0 0);"
                  >
                    <svg
                      class="m_5662a89a mantine-Rating-starSymbol"
                      data-filled="true"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="m_21342ee4 mantine-Rating-label"
                  data-read-only="true"
                >
                  <div
                    class="m_fae05d6a mantine-Rating-symbolBody"
                  >
                    <svg
                      class="m_5662a89a mantine-Rating-starSymbol"
                      data-filled="true"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="m_61734bb7 mantine-Rating-symbolGroup"
              >
                <div
                  class="m_21342ee4 mantine-Rating-label"
                  data-read-only="true"
                  style="--rating-item-z-index: 0;"
                >
                  <div
                    class="m_fae05d6a mantine-Rating-symbolBody"
                    style="--rating-symbol-clip-path: inset(0 100% 0 0);"
                  >
                    <svg
                      class="m_5662a89a mantine-Rating-starSymbol"
                      data-filled="true"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="m_21342ee4 mantine-Rating-label"
                  data-read-only="true"
                >
                  <div
                    class="m_fae05d6a mantine-Rating-symbolBody"
                  >
                    <svg
                      class="m_5662a89a mantine-Rating-starSymbol"
                      data-filled="true"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="m_61734bb7 mantine-Rating-symbolGroup"
              >
                <div
                  class="m_21342ee4 mantine-Rating-label"
                  data-read-only="true"
                  style="--rating-item-z-index: 0;"
                >
                  <div
                    class="m_fae05d6a mantine-Rating-symbolBody"
                    style="--rating-symbol-clip-path: inset(0 100% 0 0);"
                  >
                    <svg
                      class="m_5662a89a mantine-Rating-starSymbol"
                      data-filled="true"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="m_21342ee4 mantine-Rating-label"
                  data-read-only="true"
                >
                  <div
                    class="m_fae05d6a mantine-Rating-symbolBody"
                  >
                    <svg
                      class="m_5662a89a mantine-Rating-starSymbol"
                      data-filled="true"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="m_61734bb7 mantine-Rating-symbolGroup"
              >
                <div
                  class="m_21342ee4 mantine-Rating-label"
                  data-read-only="true"
                  style="--rating-item-z-index: 2;"
                >
                  <div
                    class="m_fae05d6a mantine-Rating-symbolBody"
                    style="--rating-symbol-clip-path: inset(0 50% 0 0);"
                  >
                    <svg
                      class="m_5662a89a mantine-Rating-starSymbol"
                      data-filled="true"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="m_21342ee4 mantine-Rating-label"
                  data-read-only="true"
                >
                  <div
                    class="m_fae05d6a mantine-Rating-symbolBody"
                  >
                    <svg
                      class="m_5662a89a mantine-Rating-starSymbol"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="m_61734bb7 mantine-Rating-symbolGroup"
              >
                <div
                  class="m_21342ee4 mantine-Rating-label"
                  data-read-only="true"
                  style="--rating-item-z-index: 0;"
                >
                  <div
                    class="m_fae05d6a mantine-Rating-symbolBody"
                    style="--rating-symbol-clip-path: inset(0 100% 0 0);"
                  >
                    <svg
                      class="m_5662a89a mantine-Rating-starSymbol"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="m_21342ee4 mantine-Rating-label"
                  data-read-only="true"
                >
                  <div
                    class="m_fae05d6a mantine-Rating-symbolBody"
                  >
                    <svg
                      class="m_5662a89a mantine-Rating-starSymbol"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <p
              class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              style="font-weight: 500;"
            >
              3.6
            </p>
          </div>
        </div>
        <p
          class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
        >
          A deep book about deep topics
        </p>
      </div>
      <div
        class="_paperBackground_e10025 m_1b7284a3 mantine-Paper-root"
        style="--paper-radius: var(--mantine-radius-md); --paper-shadow: var(--mantine-shadow-sm); margin-block: var(--mantine-spacing-xl);"
      >
        <style
          data-mantine-styles="inline"
        >
          .__m__-r40{--grid-gutter:var(--mantine-spacing-md);}
        </style>
        <div
          class="m_410352e9 mantine-Grid-root __m__-r40"
        >
          <div
            class="m_dee7bd2f mantine-Grid-inner"
          >
            <style
              data-mantine-styles="inline"
            >
              .__m__-r42{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
            </style>
            <div
              class="m_96bdd299 mantine-Grid-col __m__-r42"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-weight: 700;"
              >
                Title
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              >
                A title
              </p>
            </div>
            <style
              data-mantine-styles="inline"
            >
              .__m__-r46{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
            </style>
            <div
              class="m_96bdd299 mantine-Grid-col __m__-r46"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-weight: 700;"
              >
                Author
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              >
                A name
              </p>
            </div>
            <style
              data-mantine-styles="inline"
            >
              .__m__-r4a{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
            </style>
            <div
              class="m_96bdd299 mantine-Grid-col __m__-r4a"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-weight: 700;"
              >
                Publisher
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              >
                A publisher
              </p>
            </div>
            <style
              data-mantine-styles="inline"
            >
              .__m__-r4e{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
            </style>
            <div
              class="m_96bdd299 mantine-Grid-col __m__-r4e"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-weight: 700;"
              >
                Genres
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              >
                A genre
              </p>
            </div>
            <style
              data-mantine-styles="inline"
            >
              .__m__-r4i{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
            </style>
            <div
              class="m_96bdd299 mantine-Grid-col __m__-r4i"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-weight: 700;"
              >
                Rating
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              >
                3.555
              </p>
            </div>
            <style
              data-mantine-styles="inline"
            >
              .__m__-r4m{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
            </style>
            <div
              class="m_96bdd299 mantine-Grid-col __m__-r4m"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-weight: 700;"
              >
                Pages
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              >
                1000
              </p>
            </div>
            <style
              data-mantine-styles="inline"
            >
              .__m__-r4q{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
            </style>
            <div
              class="m_96bdd299 mantine-Grid-col __m__-r4q"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-weight: 700;"
              >
                Format
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              >
                Paper
              </p>
            </div>
            <style
              data-mantine-styles="inline"
            >
              .__m__-r4u{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
            </style>
            <div
              class="m_96bdd299 mantine-Grid-col __m__-r4u"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-weight: 700;"
              >
                Characters
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              >
                A character
              </p>
            </div>
            <style
              data-mantine-styles="inline"
            >
              .__m__-r52{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
            </style>
            <div
              class="m_96bdd299 mantine-Grid-col __m__-r52"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-weight: 700;"
              >
                ISBN
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              >
                999999999
              </p>
            </div>
            <style
              data-mantine-styles="inline"
            >
              .__m__-r56{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
            </style>
            <div
              class="m_96bdd299 mantine-Grid-col __m__-r56"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-weight: 700;"
              >
                Language
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              >
                Swedish
              </p>
            </div>
            <style
              data-mantine-styles="inline"
            >
              .__m__-r5a{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
            </style>
            <div
              class="m_96bdd299 mantine-Grid-col __m__-r5a"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-weight: 700;"
              >
                Series
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              >
                Title #1
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
